### PR TITLE
(MAINT) Update the changelog for the 0.5.0 release

### PR DESCRIPTION
**Pull request by senior**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,19 @@
+## 0.5.0
+
+This is a feature release that also contains bug fixes.
+
+* Added a new "v2" API that is based on the [Jookia](https://jolokia.org) library. Jookia
```
Comment (senior) [inline CHANGELOG.md]: Should I say more here? Deprecate the v1 API?

Comment (cprice404) [inline CHANGELOG.md]: I don't think we need to deprecate it yet.  There are a couple of typos above, though, `Jolokia` vs. `Jookia`.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,19 @@
...
+  provides a superset of functionality found in the current "v1" metrics API. Refer to 'document/configureation.md'
+  for more information on setting up Jookia with Trapperkeeper. The [Jookia Documentation](https://jolokia.org/documentation.html)
+  has the specifics on the new metrics API.
+
+* Ensure the JSON request is fully parsed before closing the input stream of the request
```
Comment (cprice404) [inline CHANGELOG.md]: I know we haven't been doing it in this changelog, so, not the end of the world... but for this issue and the Jolokia one it'd  be nice to have a link to the Jira tickets.

**Comment by cprice404** [comment]
If we're bumping the 'y' (which I agree seems like the right choice) then you'll make the release process  easier if you go ahead and bump the version to `0.5.0-SNAPSHOT` in the `project.clj`.

File: CHANGELOG.md (hunk 2 above)
Comment (cprice404) [inline CHANGELOG.md]: e.g. https://github.com/puppetlabs/trapperkeeper/blob/master/CHANGELOG.md

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,19 @@
+## 0.5.0
+
+This is a feature release that also contains bug fixes.
+
+* [TK-404](https://tickets.puppetlabs.com/browse/TK-404) - Added a new "v2" API that is based on the [Jolokia](https://jolokia.org) library. Jolokia
+  provides a superset of functionality found in the current "v1" metrics API. Refer to 'document/configureation.md'
+  for more information on setting up Jolokia with Trapperkeeper. The [Jolokia Documentation](https://jolokia.org/documentation.html)
```
Comment (Sharpie) [inline CHANGELOG.md]: This subsection of the docs might be better for folks who want to dive right into making requests to the API: https://jolokia.org/reference/html/protocol.html

**Comment by cprice404** [comment]
Added a commit that includes @aperiodic 's outstanding PR, plus fixes a typo and takes @Sharpie 's suggestion about the jolokia docs link.

**Pull request merged**